app/javascript/src/Dashboard/components/BackendsWidget.jsx

```diff
@@ -21,10 +43,114 @@ type Props = {
 }
 
 const BackendsWidget = (props: Props) => {
-  console.log(props)
+  console.log('what are the props' + JSON.stringify(props))
+  console.log('what are the props 2' + JSON.stringify(props.backends[0].name))
+
+  const [ isOpenArray, setIsOpenArray ] = useState([])
+
+  function handleChange (e) {
+    const item = e.target.id
+    var indexOfItem = isOpenArray.indexOf(item)
+
+    if (indexOfItem !== -1) {
+      var array = [...isOpenArray]
+      if (indexOfItem === 0) {
+        array.shift()
+        setIsOpenArray(array)
+      } else {
+        var newArray = array.splice(indexOfItem, 1)
+        setIsOpenArray(newArray)
+      }
+    } else {
+      setIsOpenArray([item, ...isOpenArray])
+    }
+  }
+
+  const APIDataListItem = props.backends.map((api, index) => {
+    let dateUpdatedAt = new Date(api.updated_at)
+
+    return (
+      <DataListItem aria-labelledby="single-action-item1">
+        <DataListItemRow>
+          <DataListItemCells
+            dataListCells={[
+              <DataListCell key="primary content">
+                <a href={api.link} id="single-action-item1">
+                  {api.name}
+                </a>
+              </DataListCell>,
+              <DataListCell key="secondary content" className="dashboard-list-secondary">
+                {dateUpdatedAt.toUTCString()}
+              </DataListCell>
+            ]}
+          />
+          <DataListAction
+            aria-labelledby="multi-actions-item1 multi-actions-action1"
+            id="actions-menu"
+            aria-label="Actions"
+            isPlainButtonAction
+          >
+            <Dropdown
+              isPlain
+              id="actions-menu"
+              position={DropdownPosition.right}
+              isOpen={isOpenArray.indexOf(`toggle-${index}`) !== -1}
+              className="dashboard-list-item-action"
+              onClick={handleChange}
+              toggle={<KebabToggle id={`toggle-${index}`} />}
+              dropdownItems={[
+                <DropdownItem key={`link-${index}`} href={api.links[0].path}>
+                  Edit
+                </DropdownItem>,
+                <DropdownItem key={`link-${index}`} href={api.links[1].path}>
+                  Overview
+                </DropdownItem>,
+                <DropdownItem key={`link-${index}`} href={api.links[2].path}>
+                  Analytics
+                </DropdownItem>,
+                <DropdownItem key={`link-${index}`} href={api.links[3].path}>
+                  Methods and Metrics
+                </DropdownItem>,
+                <DropdownItem key={`link-${index}`} href={api.links[4].path}>
+                  Mapping Rules
+                </DropdownItem>
+              ]}
```

Comment:
```suggestion
              dropdownItems={api.links.map((name, path) => (
                <DropdownItem key={name} href={path}>{name}</DropdownItem>
              ))}
```

Reply:
ahh im getting an error here when I add this and try clicking on the button, i ll look into it more

Reply:
Ah! That's because I forgot to add the curly braces: 
```diff
-              dropdownItems={api.links.map((name, path) => (
+              dropdownItems={api.links.map(({ name, path }) => (
```
I'll edit the suggestion.